packages/shared/src/hooks/search/useSearchResultsLayout.ts

```diff
@@ -0,0 +1,36 @@
+import { useViewSize, ViewSize } from '../useViewSize';
+import { useConditionalFeature } from '../useConditionalFeature';
+import { feature } from '../../lib/featureManagement';
+import { AllFeedPages } from '../../lib/query';
+import { SharedFeedPage } from '../../components/utilities';
+
+interface UseSearchResultsLayoutProps {
+  feedName?: AllFeedPages;
+}
+
+interface SearchResultsLayout {
+  isSearchResultsUpgrade: boolean;
+}
+
+export const useSearchResultsLayout = (
+  props?: UseSearchResultsLayoutProps,
+): SearchResultsLayout => {
+  const { feedName } = props ?? {};
+  const isLaptop = useViewSize(ViewSize.Laptop);
+  const { value: isSearchUpgradeExperiment, isLoading } = useConditionalFeature(
+    {
+      feature: feature.searchResultsUpgrade,
+      shouldEvaluate:
+        !!isLaptop && (!feedName || feedName === SharedFeedPage.Search),
```

Comment:
Why `!feedName`?

Reply:
because props was optional but I'm making them mandatory for consistency and for avoiding errors!

Reply:
Should we just make it mandatory then or load it from `ActiveFeedContext`, to make sure hook behaves the same in all places?

Reply:
Cool! I've switched to 

```ts
  const { feedName } = useActiveFeedNameContext();
```